plugins/vendor/webex/batch/processors/KWebexAPIDropFolderEngine.php

```diff
@@ -272,11 +297,120 @@ protected function purgeFile(KalturaDropFolderFile $dropFolderFile)
 		}
 		$this->handleFilePurged($dropFolderFile->id);
 	}
+	
+	protected function findTranscriptsForExistingEntries()
+	{
+		if (!$this->dropFolder->webexAPIVendorIntegration->siteUrl)
+		{
+			KalturaLog::info('Cannot search for transcripts, site url is not configured in vendor integration');
+			return;
+		}
+		$nextPageLink = null;
+		do
+		{
+			if ($nextPageLink)
+			{
+				$transcriptsList = $this->webexClient->sendRequestUsingDirectLink($nextPageLink);
+			}
+			else
+			{
+				$transcriptsList = $this->retrieveRecentTranscripts();
+			}
+			$nextPageLink = $this->webexClient->getNextPageLinkFromLastRequest();
+			
+			$this->handleRecentTranscriptsList($transcriptsList);
+		}
+		while ($nextPageLink);
+	}
+	
+	protected function retrieveRecentTranscripts()
+	{
+		$transcriptTimeFrameHours = kConf::getArrayValue(WebexAPIDropFolderPlugin::CONFIGURATION_TRANSCRIPT_TIME_FRAME_HOURS, WebexAPIDropFolderPlugin::CONFIGURATION_WEBEX_ACCOUNT_PARAM, WebexAPIDropFolderPlugin::CONFIGURATION_VENDOR_MAP, 12);
+		$endTime = $this->dropFolder->lastFileTimestamp;
+		$startTime = $endTime - $transcriptTimeFrameHours * 3600;
+		$transcriptsList = $this->webexClient->getTranscriptsBetweenTimes($this->dropFolder->webexAPIVendorIntegration->siteUrl, $startTime, $endTime);

```

Comment:
is there a chance that the site url will be missing?

Reply:
It is verified in the function called before, at findTranscriptsForExistingEntries()